File: LeetCode_2348/Main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var zeroFilledSubarray = function(nums) {
    let cnt = 0, ans = 0;
    for (let num of nums) {
        if (num === 0) {
            cnt++;
        } else {
            ans += (cnt * (cnt + 1)) / 2;
            cnt = 0;
        }
    }
    ans += (cnt * (cnt + 1)) / 2;
    return ans;
};

File: LeetCode_2348/Main.c
long long zeroFilledSubarray(int* nums, int numsSize) {
    long long cnt = 0, ans = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 0) {
            cnt++;
        } else {
            ans += cnt * (cnt + 1) / 2;
            cnt = 0;
        }
    }
    ans += cnt * (cnt + 1) / 2;
    return ans;
}
